src/electron/adapters/electron-storage-adapter.ts

```diff
@@ -11,13 +11,13 @@ import { Logger } from '../../common/logging/logger';
 export class ElectronStorageAdapter implements StorageAdapter {
     constructor(private readonly indexedDBInstance: IndexedDBAPI, private logger: Logger = createDefaultLogger()) {}
 
-    public setUserData(items: Object, callback?: () => void): void {
-        this.indexedDBInstance
-            .setItem(IndexedDBDataKeys.installation, items)
-            .then(() => callback && callback())
-            .catch(error => {
-                this.logger.error('Error occurred when trying to set user data: ', error);
-            });
+    public setUserData(items: Object): Promise<void> {
+        return new Promise((resolve, reject) => {
```

Comment:
This seems redundant; can it not just return `this.indexedDBInstance.setItem(...)`'s promise directly?

Reply:
so... the return type is different.. while chrome api return `Promise<void>`, `IndexedDBAPI` uses `Promise<boolean>'. I tried to cast it but is not possible and this is one way I can "convert" the promise type to match the interface type... I can see this going the other way (we make the interface have a return type of `Promise<boolean>`) but then the "convertion" will happen on the chrome adapter...

Reply:
Updated according to our offline conversation... looks much better the new way.